core/controllers/learner_group.py

```diff
@@ -0,0 +1,522 @@
+        is_valid_request = learner_group_services.is_user_a_facilitator(
+            self.user_id, learner_group_id
+        )
+
+        if not is_valid_request:
+            raise self.UnauthorizedUserException(
+                'You are not a facilitator of this learner group.')
+
+        student_ids = user_services.get_multi_user_ids_from_usernames(
+            student_usernames
+        )
+        invited_student_ids = user_services.get_multi_user_ids_from_usernames(
+            invited_student_usernames
+        )
+
+        learner_group = learner_group_services.update_learner_group(
+            learner_group_id, title, description, [self.user_id],
+            student_ids, invited_student_ids, subtopic_page_ids, story_ids
+        )
+
+        self.render_json({
+            'id': learner_group.group_id,
+            'title': learner_group.title,
+            'description': learner_group.description,
+            'facilitator_usernames': user_services.get_usernames(
+                learner_group.facilitator_user_ids),
+            'student_usernames': user_services.get_usernames(
+                learner_group.student_user_ids),
+            'invited_student_usernames': user_services.get_usernames(
+                learner_group.invited_student_user_ids),
+            'subtopic_page_ids': learner_group.subtopic_page_ids,
+            'story_ids': learner_group.story_ids
+        })
+
+    @acl_decorators.can_access_learner_groups
+    def delete(self, learner_group_id):
+        """Deletes a learner group."""
+
+        is_valid_request = learner_group_services.is_user_a_facilitator(
+            self.user_id, learner_group_id
+        )
+
+        if not is_valid_request:
+            raise self.UnauthorizedUserException(
+                'You do not have the rights to delete this learner group '
+                'as you are its facilitator.')
+
+        learner_group_services.remove_learner_group(learner_group_id)
+
+        self.render_json({
+            'success': True
+        })
+
+
+class LearnerGroupStudentProgressHandler(base.BaseHandler):
+    """Handles operations related to the learner group student's progress."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    URL_PATH_ARGS_SCHEMAS = {
+        'learner_group_id': {
+            'schema': {
+                'type': 'basestring',
+                'validators': [{
+                    'id': 'is_regex_matched',
+                    'regex_pattern': constants.LEARNER_GROUP_ID_REGEX
+                }]
+            },
+            'default_value': None
+        }
+    }
+
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {
+            'student_usernames': {
+                'schema': {
+                    'type': 'custom',
+                    'obj_type': 'JsonEncodedInString'
+                }
+            }
+        }
+    }
+
+    @acl_decorators.can_access_learner_groups
+    def get(self, learner_group_id):
```

Comment:
Can the main logic of getting all the progresses be moved to the services layer? This seems more suited tehre than here, in controllers.

Reply:
Have moved the main logic for subtopics progress to service layer as per our discussion.